### PR TITLE
#699 bug

### PR DESCRIPTION
**Pull request by wjjmjh**

### CODE REVIEW

File: tests/test_util/test_table.py [hunk 1]
```diff
@@ -1280,6 +1280,12 @@ def test_head(self):
         self.assertEqual(head.data.shape[0], 3)
         self.assertEqual(len(head.output.splitlines()), 9)
         self.assertEqual(head.data.tolist(), self.t1_rows[:3])
+        # tests when number of rows < default
+        t = make_table(data=dict(a=["a"], b=["b"]))
+        t.head()
```
Comment (GavinHuttley) [inline tests/test_util/test_table.py:1285]: these calls will be writing to stdout. Might be wise to redirect them to null by putting this in a content manager. See if pytest has a function to help with this.

File: tests/test_util/test_table.py [hunk 2]
```diff
@@ -1294,6 +1300,12 @@ def test_tail(self):
         self.assertEqual(tail.data.shape[0], 3)
         self.assertEqual(len(tail.output.splitlines()), 9)
         self.assertEqual(tail.data.tolist(), self.t1_rows[-3:])
+        # tests when number of rows < default
+        t = make_table(data=dict(a=["a"], b=["b"]))
+        t.tail()
```
Comment (GavinHuttley) [inline tests/test_util/test_table.py:1305]: same as above wrt to stdout

File: tests/test_util/test_table.py [hunk 3]
```diff
@@ -1275,17 +1276,18 @@ def test_head(self):
         display = table.display
         head = TrapOutput()
```
Comment (GavinHuttley) [inline tests/test_util/test_table.py:1277]: this is awesome, I think it means you can delete the entire TrapOutput class. Can you delete it and all uses and replace with this contextlib approach.

Comment (wjjmjh) [reply tests/test_util/test_table.py:1277]: Not a problem at all! thank you Gavin!

**Pull request merged**